### PR TITLE
Support US-ASCII encoding for posix downcase

### PR DESCRIPTION
**Pull request by prvithani**

### CODE REVIEW

File: posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c [hunk 1]
```diff
@@ -65,25 +66,29 @@ static VALUE
...
+  rb_encoding *input_str_enc;
   VALUE result;
 
   /* Check if the receiver is string. Raise `TypeError` otherwise */
   Check_Type(rb_string, T_STRING);
 
   /*
-   * Only UTF-8 encoding is supproted at the moment for posix downcase.
-   * Raises Encoding::CompatibilityError if encoding of reciever is not UTF-8.
+   * Only UTF-8 and US-ASCII encoding are supproted at the moment for posix
```
Comment (OPhamster) [inline posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c]:
```suggestion
   * Only UTF-8 and US-ASCII encoding are supported at the moment for posix
```

File: posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c [hunk 2]
```diff
@@ -24,11 +24,30 @@ static rb_encoding *utf8_enc = NULL;
...
+posix_downcase_usascii(const char *string, size_t len)
+{
+  char *lower_str;
+  size_t curr_char;
+  int c;
+  lower_str = (char *) malloc(len);
+  for (curr_char = 0; curr_char < len; curr_char++)
+    {
+      c = string[curr_char];
+      lower_str[curr_char] = (c >= 'A' && c <= 'Z') ? c + ('a' - 'A') : c;
```
Comment (prvithani) [inline posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c:41]: @pritambaral: Although we discussed this, I was wondering why not use `tolower` instead which I assume does the same thing?

File: posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c [hunk 3]
```diff
@@ -1,4 +1,3 @@
-#include <wctype.h>
```
Comment (pritambaral) [inline posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c:1]: How does it compile without this line? Did you check on a Linux system?

File: posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c (hunk 2 above)
Comment (pritambaral) [reply posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c:41]: `tolower` is not the same as this, but the difference matters only for non-ASCII characters.

File: posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c [hunk 4]
```diff
@@ -24,11 +24,30 @@ static rb_encoding *utf8_enc = NULL;
...
+ * Downcases the supplied string according to US-ASCII semantics.
+ * Returns a freshly allocated string that the caller is responsible for.
  */
 static char *
-posix_downcase(const char *string, size_t len)
+posix_downcase_usascii(const char *string, size_t len)
+{
+  char *lower_str;
+  size_t curr_char;
+  int c;
```
Comment (pritambaral) [inline posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c]:
```suggestion
  unsigned char c;
```

File: posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c (hunk 3 above)
Comment (pritambaral) [inline posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c]: How does it compile without this line? Did you check on a Linux system?

**Review by pritambaral — CHANGES_REQUESTED** [review]

File: posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c (hunk 3 above)
Comment (prvithani) [reply posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c]: Ah removal of this line was unintentional. Although, it compiles with warning.

**Pull request merged**